plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/frontend.ts

```diff
@@ -156,17 +167,46 @@ const { actions, state } = store< VariableProductAddToCartWithOptionsStore >(
 				if ( ! context ) {
 					return true;
 				}
-				const { availableVariations, selectedAttributes } = context;
+				const {
+					availableVariations,
+					selectedAttributes,
+					quantity,
+					productId,
+					childProductId,
+					productType,
+				} = context;
 
-				const matchedVariation = getMatchedVariation(
+				const productObject = getProductData(
+					childProductId || productId,
+					productType,
 					availableVariations,
 					selectedAttributes
 				);
 
+				if ( ! productObject?.id ) {
+					return false;
+				}
+
+				const { max: productMaxCartQty, id } = productObject;
+
+				const cartItems = wooState.cart?.items ?? [];
+
+				if ( ! productMaxCartQty ) {
+					return true;
+				}
+
+				const variableProductQty =
+					cartItems.find( ( item ) => item.id === id )?.quantity || 0;
+				const maxCartQty = productMaxCartQty || 0;
+
+				// Returns form quantity selector component value.
+				const qty = quantity[ id ?? productId ];
```

Comment:
I think `id` will always exist because we checked it in:

```TS
				if ( ! productObject?.id ) {
					return false;
				}
```

so this line could be simplified to:

```suggestion
				const qty = quantity[ id ];
```